### ISSUE TITLE
Exceptions are rethrown when using fallback of CheckedConsumer and CheckedRunnable

### ISSUE THREAD

**Issue by AvaPL** [jhalterman/failsafe#206]
Here is the code, I think it's self-explanatory:

```java
//#of(R fallback) - works as expected.
Failsafe.with(Fallback.of(0)).run(() -> {
    throw new RuntimeException();
});

//#of(CheckedSupplier<? extends R> fallback) - works as expected.
Failsafe.with(Fallback.of(() -> 0)).run(() -> {
    throw new RuntimeException();
});

//#of(CheckedFunction<ExecutionAttemptedEvent<? extends R>,? extends R> fallback) - works as expected.
Failsafe.with(Fallback.of(executionAttemptedEvent -> 0)).run(() -> {
    throw new RuntimeException();
});

//#of(CheckedConsumer<ExecutionAttemptedEvent<? extends R>> fallback) - rethrows exception.
try {
    Failsafe.with(Fallback.of(executionAttemptedEvent -> {})).run(() -> {
        throw new RuntimeException();
    });
} catch (RuntimeException e) {
    System.err.println(e + " for CheckedConsumer caught");
}

//#of(CheckedRunnable fallback) - rethrows exception.
try {
    Failsafe.with(Fallback.of(() -> {})).run(() -> {
        throw new RuntimeException();
    });
} catch (RuntimeException e) {
    System.err.println(e + " for CheckedRunnable caught");
}
```

Output:

> java.lang.RuntimeException for CheckedConsumer caught
> java.lang.RuntimeException for CheckedRunnable caught

It's problematic when I just want to suppress exception and just print a message.

**Comment by AvaPL** [comment]
Similar issue happens when I want to use a fallback with policy and `Failsafe::run` eg. `Failsafe.with(fallback, policy).run(method);` In this case no matter what callback is used (even those returning a value) an exception is rethrown. It's really counterintuitive to have a callback that executes given method for handled exception AND rethrows the exception.

**Comment by jhalterman** [comment]
This looks like a recent regression. Thanks for filing.

**Comment by jhalterman** [comment]
> In this case no matter what callback is used (even those returning a value) an exception is rethrown

You mean no matter what fallback is used?

I think 7fa504c651fbb9652a1d988e8a6d46da16f57442 fixes the case you described. Let me know if not.

**Comment by jhalterman** [comment]
Released in 2.3.1.
